libopae/plugins/xfpga/metrics/metrics_utils.c

```diff
@@ -826,19 +817,32 @@ fpga_result enum_fpga_metrics(fpga_handle handle)
 			// MCP
 		case FPGA_HW_MCP: {
 
-			result = enum_powermgmt_metrics(&(_handle->fpga_enum_metric_vector), &metric_num, _token->sysfspath, FPGA_HW_MCP);
-			if (result != FPGA_OK) {
-				FPGA_MSG("Failed to Enum Power metrics.");
+			memset_s(metrics_path, SYSFS_PATH_MAX, 0);
+
+			if (sysfs_get_fme_pwr_path(_token, metrics_path) == FPGA_OK) {
+
+				result = enum_powermgmt_metrics(&(_handle->fpga_enum_metric_vector), &metric_num, metrics_path, FPGA_HW_MCP);
+				if (result != FPGA_OK) {
+					FPGA_MSG("Failed to Enum Power metrics.");
+				}
 			}
 
-			result = enum_thermalmgmt_metrics(&(_handle->fpga_enum_metric_vector), &metric_num, _token->sysfspath, FPGA_HW_MCP);
-			if (result != FPGA_OK) {
-				FPGA_MSG("Failed to Enum Thermal metrics.");
+			memset_s(metrics_path, SYSFS_PATH_MAX, 0);
+			if (sysfs_get_fme_temp_path(_token, metrics_path) == FPGA_OK) {
+
+				result = enum_thermalmgmt_metrics(&(_handle->fpga_enum_metric_vector), &metric_num, metrics_path, FPGA_HW_MCP);
+				if (result != FPGA_OK) {
+					FPGA_MSG("Failed to Enum Thermal metrics.");
+				}
 			}
 
-			result = enum_perf_counter_metrics(&(_handle->fpga_enum_metric_vector), &metric_num, _token->sysfspath, FPGA_HW_MCP);
-			if (result != FPGA_OK) {
-				FPGA_MSG("Failed to Enum Perforamnce metrics.");
+			memset_s(metrics_path, SYSFS_PATH_MAX, 0);
+			if (sysfs_get_fme_perf_path(_token, metrics_path) == FPGA_OK) {
+
+				result = enum_perf_counter_metrics(&(_handle->fpga_enum_metric_vector), &metric_num, metrics_path, FPGA_HW_MCP);
+				if (result != FPGA_OK) {
+					FPGA_MSG("Failed to Enum Perforamnce metrics.");
```

Comment:
Seems like an error condition.  Use FPGA_ERR and return an error code?

Reply:
fixed